### ISSUE TITLE
auto-fix generates invalid jsx

### ISSUE THREAD

**Issue by stefanprobst** [microsoft/vscode-eslint#1755]
with [two rules](https://github.com/stefanprobst/issue-eslint-react/blob/main/.vscode/settings.json#L6-L7) from `eslint-plugin-react` enabled, i get invalid jsx when either selecting "Fix all auto-fixable problems" or with `Ctrl+S` and `"source.fixAll.eslint": "explicit"`.

see this repo for a minimal reproduction: https://github.com/stefanprobst/issue-eslint-react (i did check in a fresh vscode install with only the eslint extension installed)

steps:
- clone [repo](https://github.com/stefanprobst/issue-eslint-react)
- run `pnpm install`
- open `src/app.tsx` in vscode and press Ctrl+S
- see invalid jsx

some observations:

- running `pnpm run lint --fix` from the terminal does *not* have this problem. the resulting (correct) jsx is:
  ```tsx
  <button aria-disabled={true} className="test">Test</button>
  ```
- after `Ctrl+S` in vscode, the resulting jsx is:
  ```tsx
  <button aria-disabled className="test"={true}>Test</button>
  ```

**Comment by dbaeumer** [comment]
Dups #1745

Although different rules there was an underlying problem with generating all fixes which got solved in #1745

**Comment by stefanprobst** [comment]
@dbaeumer when do you expect the fix for this issue to be released? thanks
